config.json

```diff
@@ -1135,6 +1135,24 @@
           "strings"
         ]
       },
+      {
+        "slug": "killer-sudoku-helper",
+        "name": "Killer Sudoku Helper",
+        "uuid": "8c45a47d-b3e3-484c-a124-90136ec838fd",
+        "practices": [
+          "constructors",
+          "lists",
+          "recursion"
+        ],
+        "prerequisites": [
+          "basics",
+          "numbers",
+          "loops",
+          "arrays",
+          "if-else-statements"
+        ],
```

Comment:
- The prerequisites should list existing concepts (you can find them in the [concepts](https://github.com/exercism/java/tree/main/concepts) directory). I think `loops` should therefore be `for-loops`.
- There is no need to add "transitive" dependencies: the `basics` concept is a prerequisite of other concepts like `numbers`, so you only need to list the concept the exercise directly depends on.
- For practice exercises we normally add the concepts from the `practices` array in the `prerequisites` array too. That way students first have to complete the concept exercise for a concept. Practice exercises can then be used to advance their knowledge in the concept. The exception here is when a concept isn't required to complete the exercise. For example, you mentioned that this exercise is _probably_ going to be solved using recursion, so it sounds like that wouldn't be a prerequisite since there are other ways of solving it.

Reply:
Taking all of this in mind and looking at [the current syllabus](https://exercism.org/tracks/java/concepts) and how the concepts depend on each other, I think we can make do with the following prerequisites:

- numbers
- lists

Reply:
Thanks a lot for the detailed explanation. So at the moment this would look like 


```
        "practices": [
          "for-loops"
        ],
        "prerequisites": [
          "basics",
          "numbers"
        ],
```

Reply:
Hmm no, that is not what I meant. I was thinking more like

```json
{
  "practices": [
    "for-loops",
    "if-else-statements",
    "numbers",
    "lists"
  ],
  "prerequisites": [
    "numbers",
    "lists"
  ]
}
```